korma/src/commonMain/kotlin/com/soywiz/korma/geom/IRectangle.kt

```diff
@@ -0,0 +1,92 @@
+package com.soywiz.korma.geom
+
+interface IRectangle {
+    val x: Double
+    val y: Double
+    val width: Double
+    val height: Double
+
+    companion object {
+        inline operator fun invoke(
+            x: Double,
+            y: Double,
+            width: Double,
+            height: Double
+        ): IRectangle = Rectangle(x, y, width, height)
+
+        inline operator fun invoke(x: Float, y: Float, width: Float, height: Float): IRectangle =
+            Rectangle(x, y, width, height)
+
+        inline operator fun invoke(x: Int, y: Int, width: Int, height: Int): IRectangle =
+            Rectangle(x, y, width, height)
+
+        // Creates a rectangle from 2 points where the (x,y) is the top left point
+        // with the same width and height as the point.
+        //
+        // Rect XY   Point 2
+        // │        │
+        // ▼        ▼
+        // ┌────────┐
+        // │        │
+        // │        │
+        // └────────┘
+        // ▲
+        // │
+        // Point 1
+        operator fun invoke(point1: IPoint, point2: IPoint): Rectangle {
+            val topLeftX = minOf(point1.x, point2.x)
+            val topLeftY = minOf(point1.y, point2.y)
+            val bottomRightX = maxOf(point1.x, point2.x)
+            val bottomRightY = maxOf(point1.y, point2.y)
+            return Rectangle(topLeftX, topLeftY, bottomRightX - topLeftX, bottomRightY - topLeftY)
+        }
+    }
+}
+
+@Deprecated(
+    "Properties with underscores are deprecated and will be removed soon",
+    ReplaceWith("x"),
+    DeprecationLevel.ERROR
+)
+val IRectangle._x get() = x
```

Comment:
Yeah, we should even remove them already since we are going into 3.0 and this has been here for quite a long time already